packages/connect/src/ws_orderbook_channel.ts

```diff
@@ -55,112 +49,84 @@ export class WebSocketOrderbookChannel implements OrderbookChannel {
      *                               channel updates
      */
     public subscribe(subscriptionOpts: OrderbookChannelSubscriptionOpts, handler: OrderbookChannelHandler): void {
-        assert.doesConformToSchema(
-            'subscriptionOpts',
+        assert.isOrderbookChannelSubscriptionOpts('subscriptionOpts', subscriptionOpts);
+        assert.isOrderbookChannelHandler('handler', handler);
+        const newSubscription: Subscription = {
             subscriptionOpts,
-            schemas.relayerApiOrderbookChannelSubscribePayload,
-        );
-        assert.isFunction('handler.onSnapshot', _.get(handler, 'onSnapshot'));
-        assert.isFunction('handler.onUpdate', _.get(handler, 'onUpdate'));
-        assert.isFunction('handler.onError', _.get(handler, 'onError'));
-        assert.isFunction('handler.onClose', _.get(handler, 'onClose'));
-        this._subscriptionCounter += 1;
+            handler,
+        };
+        this._subscriptions.push(newSubscription);
         const subscribeMessage = {
             type: 'subscribe',
             channel: 'orderbook',
-            requestId: this._subscriptionCounter,
+            requestId: this._subscriptions.length - 1,
```

Comment:
Should we increment here or use a random hash instead? I'm concerned about how unsubscribing, then re-subscribing would result in the same `requestId` and the backend server would think it was communicating with the same subscription...

Reply:
This is changing to a UUID in v2 of SRA.